### PR TITLE
Update `Accounting Integration` API docs

### PR DESCRIPTION
**Pull request by arjun-rathee-sap**
- Modify sample API payloads and responses to be semantically accurate
- Update API descriptions for List and Vendor endpoints to clarify the scope of operations
- Update swagger schema to indicate required attributes

### CODE REVIEW

**Comment by arjun-rathee-sap** [comment]
@jonikamelcher - requesting a review to update the preview docs for Accounting Integration. Please let me know if you have any questions/concerns with the changes

File: src/api-reference/accounting-integration/v4.accountingintegration-vendor.markdown [hunk 1]
```diff
@@ -13,6 +13,8 @@ This endpoint allows users to add, update, or delete ERP vendor data that is cac
 
 In addition, events will be published to notify API users of important updates. For more information, please see [Accounting Integration Events](/event-topics/event-subscription-topic-accountingintegration.html).
 
+Creating Vendors through this API will not automatically import those vendors into SAP Concur. Customers will need to add the accounting and employee vendors in SAP Concur via the provided UI screens.
```
Comment (jonikamelcher) [inline src/api-reference/accounting-integration/v4.accountingintegration-vendor.markdown:16]: Hi @arjun-rathee-sap. Could you please move this to the Limitations section? Thank you.

Comment (arjun-rathee-sap) [reply src/api-reference/accounting-integration/v4.accountingintegration-vendor.markdown]: Done

**Review by jonikamelcher — APPROVED** [review]

**Comment by jonikamelcher** [comment]
:shipit:

**Pull request merged**